tests/testTransforms.py

```diff
@@ -0,0 +1,468 @@
+        sip = lsst.meas.astrom.SipForwardTransform.convert(scaled)
+        self.checkTransformEquivalence(sip, scaled)
+
+    def testMakeWcs(self):
+        fwd = self.makeRandom()
+        rev = SipReverseTransform(  # this isn't actually the inverse of fwd, but that doesn't matter
+            fwd.getPixelOrigin(),
+            fwd.getCDMatrix(),
+            makeRandomPolynomialTransform(4)
+        )
+        crval = lsst.afw.geom.Point2D(45.0, 45.0)
+        wcs = lsst.meas.astrom.makeWcs(
+            fwd, rev,
+            lsst.afw.coord.IcrsCoord(crval, lsst.afw.geom.degrees)
+        )
+        def t1(p):
+            sky = lsst.afw.coord.IcrsCoord(crval + lsst.afw.geom.Extent2D(p), lsst.afw.geom.degrees)
+            return rev(wcs.skyToIntermediateWorldCoord(sky))
+        def t2(p):
+            sky = lsst.afw.coord.IcrsCoord(crval + lsst.afw.geom.Extent2D(p), lsst.afw.geom.degrees)
+            return wcs.skyToPixel(sky)
+        self.checkTransformEquivalence(t1, t2)
+
+class SipReverseTransformTestCase(lsst.utils.tests.TestCase, TransformTestMixin):
+
+    def setUp(self):
+        numpy.random.seed(50)
+
+    def makeRandom(self):
+        return makeRandomSipReverseTransform(4)
+
+    def testConstruction(self):
+        poly = makeRandomPolynomialTransform(4)
+        cd = lsst.afw.geom.LinearTransform(numpy.random.randn(2,2))
+        crpix = lsst.afw.geom.Point2D(*numpy.random.randn(2))
+        sip = SipReverseTransform(crpix, cd, poly)
+        offset = lsst.afw.geom.Extent2D(crpix)
+        cdInverse = cd.invert()
+        self.checkTransformEquivalence(
+            sip,
+            lambda p: offset + lsst.afw.geom.Extent2D(cdInverse(p)) + poly(cdInverse(p))
+        )
+
+    def testConvertPolynomial(self):
+        poly = makeRandomPolynomialTransform(4)
+        cd = lsst.afw.geom.LinearTransform(numpy.random.randn(2,2))
+        crpix = lsst.afw.geom.Point2D(*numpy.random.randn(2))
+        sip = lsst.meas.astrom.SipReverseTransform.convert(poly, crpix, cd)
+        self.checkTransformEquivalence(sip, poly)
+
+    def testConvertScaledPolynomialManual(self):
+        scaled = makeRandomScaledPolynomialTransform(4)
+        cd = lsst.afw.geom.LinearTransform(numpy.random.randn(2,2))
+        crpix = lsst.afw.geom.Point2D(*numpy.random.randn(2))
+        sip = lsst.meas.astrom.SipReverseTransform.convert(scaled, crpix, cd)
+        self.checkTransformEquivalence(sip, scaled)
+
+    def testConvertScaledPolynomialAutomatic(self):
+        scaled = makeRandomScaledPolynomialTransform(4)
+        sip = lsst.meas.astrom.SipReverseTransform.convert(scaled)
+        self.checkTransformEquivalence(sip, scaled)
+
+
+class ScaledPolynomialTransformFitterTestCase(lsst.utils.tests.TestCase):
+
+    def setUp(self):
+        numpy.random.seed(50)
+
+    def testFromMatches(self):
+        # Setup artifical matches that correspond to a known (random) PolynomialTransform.
+        order = 3
```

Comment:
I'd rather not; it'd only be used in one place, and I think the comments are sufficient to make it clear where in the code we're doing prep work rather than doing the tests.  And I don't want to comb through this function to find which variables I set in the top half I'll need to pass out of the function unless there's a very good reason to split them up.